nuxeo-features/nuxeo-platform-userworkspace/nuxeo-platform-userworkspace-core/src/main/java/org/nuxeo/ecm/platform/userworkspace/core/service/AbstractUserWorkspaceImpl.java

```diff
@@ -250,11 +269,11 @@ protected PathRef getExistingUserWorkspace(CoreSession session, PathRef rootref,
             // try next name
         }
         if (freeRef != null) {
-            PathRef ref = freeRef; // effectively final
-            CoreInstance.doPrivileged(session, s -> {
-                doCreateUserWorkspace(s, ref, principal, username);
+            DocumentModel uwsModel = doCreateUserWorkspace(session, freeRef, username);
+            DocumentModel uwsDoc = CoreInstance.doPrivileged(session, s -> {
+                return s.getOrCreateDocument(uwsModel, doc -> initCreateUserWorkspace(s, doc, username));
```

Comment:
Return the path and not the `DocumentModel`, as we're only interested in the path.